tools/autodeployment/builds.js

```diff
@@ -1,13 +1,26 @@
 const { logger, createError } = require('@senecacdot/satellite');
 const shell = require('shelljs');
 const mergeStream = require('merge-stream');
-const streamBuffers = require('stream-buffers');
+const Cache = require('streaming-cache');
+
+// Create an LRU cache of build log streams, which we'll key on git sha
+const buildLogCache = new Cache();
 
 class Build {
   constructor(type, githubData) {
     this.type = type;
     this.githubData = githubData;
     this.startedDate = new Date();
+
+    // Remember the git SHA for this build
+    this.sha = githubData.after;
```

Comment:
I think this won't work in production. I left a [comment in 2650](https://github.com/Seneca-CDOT/telescope/issues/2650#issuecomment-1018045662) for a possible fix.